src/mount/conf.cc

```diff
@@ -0,0 +1,92 @@
+// -*- mode:C++; tab-width:8; c-basic-offset:2; indent-tabs-mode:t -*-
+// vim: ts=8 sw=2 smarttab
+
+#include "common/ceph_context.h"
+#include "common/ceph_argparse.h"
+#include "global/global_init.h"
+#include "common/config.h"
+#include "auth/KeyRing.h"
+#include "mount.ceph.h"
+#include "mon/MonClient.h"
+
+#include <iostream>
+
+extern "C" void mount_ceph_get_config_info(const char *config_file,
+					   const char *name,
+					   struct ceph_config_info *cci)
+{
+  int err;
+  KeyRing keyring;
+  CryptoKey secret;
+  std::string secret_str;
+  std::string monaddrs;
+  vector<const char *> args = { "--name", name };
+  bool first = true;
+
+  if (config_file) {
+    args.push_back("--conf");
+    args.push_back(config_file);
+  }
+
+  /* Create CephContext */
+  auto cct = global_init(NULL, args, CEPH_ENTITY_TYPE_CLIENT,
+			 CODE_ENVIRONMENT_UTILITY,
+			 CINIT_FLAG_NO_DAEMON_ACTIONS|CINIT_FLAG_NO_MON_CONFIG);
+  auto& conf = cct->_conf;
+
+  conf.parse_env(cct->get_module_type()); // environment variables override
+  conf.apply_changes(nullptr);
+
+  MonClient monc = MonClient(cct.get());
+  err = monc.build_initial_monmap();
+  if (err)
+    goto scrape_keyring;
+
+  for (auto p = monc.monmap.addr_mons.begin(); p != monc.monmap.addr_mons.end(); ++p) {
```

Comment:
could use range-based loop, like
```c++
for (const auto& mon : monc.monmap.addr_mons) {
  auto& eaddr = p.first;
  // ...
}
```

Reply:
What's the benefit of doing so?

Reply:
i think the benefit is better readability and less repeating.

Reply:
Fair enough. Changed to range-based loop.